### PR TITLE
result_type

### PR DESCRIPTION
**Pull request by KevinUnadkat**
Hey. @djl11 I am doing sub-task of #18

### CODE REVIEW

**Comment by KevinUnadkat** [comment]
hey @djl11 
pks add "pytorch frontend" tag to this pr

**Comment by KevinUnadkat** [comment]
Hey @guillesanbri

**Comment by guillesanbri** [comment]
Hi! Thanks a lot for contributing, I'll take a look now. It seems like the Issue has not been correctly linked to the ToDo task list, please do so asap to avoid the situation where other people assign themselves the same function :slightly_smiling_face:

[Docs on how to do this.](https://lets-unify.ai/ivy/contributing/the_basics.html#todo-list-issues)

**Comment by KevinUnadkat** [comment]
hey @guillesanbri 

I'm having an issue with configuring python test in my vscode  . I'm sharing an screen shot of the error I'm getting pls do check it , pls help me with it.

u can read  about the error in bottom right of the ss.

Screenshot 2022-12-12 151734

**Comment by KevinUnadkat** [comment]
> Hi, unfortunately I haven't faced that error, probably best to ask about it in the vs-code channel in discord :slightly_smiling_face: 

Ok , I'll try that 
But can u give be review abt the error I'm getting while running friend test

**Comment by KevinUnadkat** [comment]
Hey @guillesanbri 
Error I'm getting during frontend test, I don't think it is because of something with my setup can u please give it a check.

File: ivy/functional/frontends/torch/utilities.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# local
+import ivy
+
+
+def result_type(tensor1, tensor2):
```
Comment (guillesanbri) [inline ivy/functional/frontends/torch/utilities.py:5]: probably missing `@to_ivy_arrays_and_back`

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+@handle_frontend_test(
+    fn_tree="torch.result_type",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("valid"),
+        valid_axis=True,
+        max_axes_size=1,
+    ),
+    keepdims=st.booleans(),
+)
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:17]: I don't think that neither `axis` nor `keepdims` are needed here

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+def test_torch_result_type(
+    dtype_x_axis,
+    keepdims,
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:20]: same as above

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x[0],
+        axis=axis,
+        keepdims=keepdims,
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:41]: these are not the args of this function so they shouldn't really be here

**Review by guillesanbri — CHANGES_REQUESTED** [review]
Hi! I've added some feedback which may help you in getting the test to pass, thanks! :slightly_smiling_face:

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py (hunk 4 above)
Comment (KevinUnadkat) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:41]: Should I remove input=x(0) also?

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py (hunk 2 above)
Comment (KevinUnadkat) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:17]: @guillesanbri axis and keepdims are mentined in this 
ivy_tests/test_ivy/test_functional/test_core/test_utility.py

that is the reason i'm using them.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py (hunk 4 above)
Comment (KevinUnadkat) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:41]: @guillesanbri do i also need remove input=x[0].

**Comment by KevinUnadkat** [comment]
@guillesanbri 
I made some changes as u said can u review them.

**Comment by KevinUnadkat** [comment]
@guillesanbri 
I made some changes as u said can u review them.
and
why test-frontend-torch is taking  very long time complete?

**Comment by KevinUnadkat** [comment]
@guillesanbri 
 frontend test showed me this error while I ran it on my pc


@handle_frontend_test(
        fn_tree="torch.result_type",
        dtype_and_x=helpers.dtype_and_values(
            available_dtypes=helpers.get_dtypes("valid"),
        ),
    )
    def test_torch_result_type(
        dtype_and_x,
        as_variable,
        with_out,
        num_positional_args,
        native_array,
        on_device,
        fn_tree,
        frontend,
    ):
      input_dtype, x = dtype_and_x,
E       ValueError: not enough values to unpack (expected 2, got 1)
E       Falsifying example: test_torch_result_type(
E           dtype_and_x=(['int8'], [array(0, dtype=int8)]),
E           num_positional_args=0,
E           as_variable=[False],
E           native_array=[False],
E           with_out=False,
E           on_device='cpu',
E           fn_tree='ivy.functional.frontends.torch.result_type',
E           frontend='torch',
E       )
E
E       You can reproduce this example by temporarily adding @reproduce_failure('6.55.0', b'AAIAAAAAAAAAAA==') as a decorator on your test case

ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:22: ValueError
============================================================ warnings summary =============================================================
../../usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19
  /usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

**Comment by KevinUnadkat** [comment]
I got this error 
Screenshot_2022-12-16-22-25-25-95_40deb401b9ffe8e1df2f1cc5ba480b12

**Comment by KevinUnadkat** [comment]
@guillesanbriy is merge block in my branch

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    num_positional_args,
+    native_array,
+    on_device,
+    fn_tree,
+    frontend,
+):
+    input_dtype, x = dtype_and_x
+    tensor1_dtype, tensor1 = input_dtype[0] , x[0]
+    tensor2_dtype, tensor2 = input_dtype[1] , x[1]
+    helpers.test_frontend_function(
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:26]: Please add `tensor1` and `tensor2` as arguments to this `test_frontend_function`

File: ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        input_dtypes=[tensor1_dtype, tensor2_dtype],
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        x=x[0],
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_utilities.py:35]: What does `x` represent?

**Comment by KevinUnadkat** [comment]
@guillesanbri @djl11 
can u pls review changes i made.

**Comment by KevinUnadkat** [comment]
@ZiadAmerr can u pls review i guess it is ready to merge

**Review by guillesanbri — CHANGES_REQUESTED** [review]
once these changes are made I think the error left is on our side so I'll merge it then :smile:

**Comment by KevinUnadkat** [comment]
@guillesanbri 
made the changes can u pls merge.

**Comment by guillesanbri** [comment]
Merged into master, thanks a lot for contributing once again! :smile:

**Pull request merged**